openquake/hazardlib/gsim/sandikkaya_akkar_2017.py

```diff
@@ -0,0 +1,176 @@
+Module exports :class:`SandikkayaAkkar2017Rhyp`
+"""
+import numpy as np
+
+from openquake.baselib.general import CallableDict
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import CAV, IA
+
+
+_compute_logarithmic_distance_term = CallableDict()
+
+
+@_compute_logarithmic_distance_term.add("rjb")
+def _compute_logarithmic_distance_term_1(kind, C, ctx):
+    """uses rjb for distance part"""
+    return (C["a4"] + C["a5"] * (ctx.mag - 6.75)) * np.log(
+        np.sqrt(ctx.rjb**2 + C["a6"] ** 2)
+    )
+
+
+@_compute_logarithmic_distance_term.add("repi")
+def _compute_logarithmic_distance_term_2(kind, C, ctx):
+    """uses repi for distance part"""
+    return (C["a4"] + C["a5"] * (ctx.mag - 6.75)) * np.log(
+        np.sqrt(ctx.repi**2 + C["a6"] ** 2)
+    )
+
+
+@_compute_logarithmic_distance_term.add("rhypo")
+def _compute_logarithmic_distance_term_3(kind, C, ctx):
+    """uses rhypo for distance part"""
+    return (C["a4"] + C["a5"] * (ctx.mag - 6.75)) * np.log(
+        np.sqrt(ctx.rhypo**2 + C["a6"] ** 2)
+    )
+
+
+def get_mean_values(kind, C, ctx):
+    """
+    Returns the mean values for a specific IMT from page 1887 of Sandikkaya and Akkar (2017) 
+    """
+    FNM = np.zeros_like(ctx.rake)
+    FNM[np.absolute(ctx.rake + 90) < 45] = 1
+    FRV = np.zeros_like(ctx.rake)
+    FRV[np.absolute(ctx.rake - 90) < 45] = 1
+    return (
+        C["a1"]
+        + C["a2"] * (ctx.mag - 6.75)
+        + C["a3"] * (8.5 - ctx.mag) ** 2
+        + _compute_logarithmic_distance_term(kind, C, ctx)
+        + C["a7"] * FNM
+        + C["a8"] * FRV
+        + C["a9"] * np.log(np.minimum(ctx.vs30, 1000) / 750)
+    )
+
+
+class SandikkayaAkkar2017Rjb(GMPE):
+    """
+    Implements ESM GMPE using rjb distances developed by M. Abdullah Sandikkaya
+    and Sinan Akkar, published as "Cumulative Absolute Velocity, Arias
+    Intensity and significant duration predictive models from a pan-European
+    strong-motion dataset" (2017, Bulletin of Earthquake Engineering,
+    Volume 15, pages 1881 - 1898).
+    """
+
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = {CAV, IA}
+
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.GEOMETRIC_MEAN
+
+    #: Supported standard deviation types are inter-event, intra-event
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = {
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT,
+    }
+
+    #: Required site parameters are Vs30, Vs30 type (measured or inferred),
+    REQUIRES_SITES_PARAMETERS = {"vs30"}
+
+    #: Required rupture parameters are
+    REQUIRES_RUPTURE_PARAMETERS = {"mag", "rake"}
+
+    #: Required distance measures are
+    REQUIRES_DISTANCES = {"rjb"}
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        [self.kind] = self.REQUIRES_DISTANCES
```

Comment:
You can remove the `__init__` method entirely and just set at class level `kind = "rjb"`